src/python/pants/backend/python/tasks/python_repl.py

```diff
@@ -3,41 +3,72 @@
 
 from pex.pex_info import PexInfo
 
+from pants.backend.python.subsystems.ipython import IPython
 from pants.backend.python.targets.python_requirement_library import PythonRequirementLibrary
 from pants.backend.python.targets.python_target import PythonTarget
 from pants.backend.python.tasks.python_execution_task_base import PythonExecutionTaskBase
+from pants.base.deprecated import resolve_conflicting_options
 from pants.base.exception_sink import ExceptionSink
 from pants.task.repl_task_mixin import ReplTaskMixin
 
 
 class PythonRepl(ReplTaskMixin, PythonExecutionTaskBase):
   """Launch an interactive Python interpreter session."""
 
+  def _resolve_conflicting_options(self, *, old_option: str, new_option: str):
+    return resolve_conflicting_options(
+      old_option=old_option,
+      new_option=new_option,
+      old_scope='repl-py',
+      new_scope='ipython',
+      old_container=self.get_options(),
+      new_container=IPython.global_instance().options,
+    )
+
   @classmethod
   def register_options(cls, register):
     super().register_options(register)
-    # TODO: Create a python equivalent of register_jvm_tool, and use that instead of these
-    # ad-hoc options.
-    register('--ipython', type=bool,
+    register('--ipython', type=bool, fingerprint=True,
+             removal_version="1.27.0.dev0",
+             removal_hint="Use `--ipython-enabled` instead of `--repl-py-ipython`.",
              help='Run an IPython REPL instead of the standard python one.')
     register('--ipython-entry-point', advanced=True, default='IPython:start_ipython',
-             help='The IPython REPL entry point.')
-    register('--ipython-requirements', advanced=True, type=list, default=['ipython==1.0.0'],
-             help='The IPython interpreter version to use.')
+             removal_version="1.27.0.dev0",
+             removal_hint="Use `--ipython-entry-point` instead of `--repl-py-ipython-entry-point`.",
+             fingerprint=True, help='The IPython REPL entry point.')
+    register('--ipython-requirements', advanced=True, type=list, default=['ipython==5.8.0'],
+             removal_version="1.27.0.dev0",
+             removal_hint="Use `--ipython-version` and `--ipython-extra-requirements` instead of "
+                          "`--repl-py-ipython-requirements`.",
+             fingerprint=True, help='The IPython interpreter version to use.')
+
+  @classmethod
+  def subsystem_dependencies(cls):
+    return super().subsystem_dependencies() + (IPython,)
 
   @classmethod
   def select_targets(cls, target):
     return isinstance(target, (PythonTarget, PythonRequirementLibrary))
 
   def extra_requirements(self):
-    if self.get_options().ipython:
-      return self.get_options().ipython_requirements
-    else:
+    if not self.get_options().ipython and not IPython.global_instance().options.enabled:
```

Comment:
I didn't use `resolve_conflicting_options()` here because we use it in a below method and I don't want us to print the same message twice.

(I also didn't comment this because it's temporary deprecation code going away very soon.)